hibernate-core/src/main/java/org/hibernate/mapping/Index.java

```diff
@@ -223,7 +224,22 @@ public String getName() {
 	}
 
 	public void setName(String name) {
-		this.name = name;
+		if (
+				StringHelper.isNotEmpty( name ) &&
+						Dialect.QUOTE.indexOf( name.charAt( 0 ) ) > -1 //TODO: deprecated, remove eventually
+				) {
+			quoted = true;
+			this.name = name.substring( 1, name.length() - 1 );
```

Comment:
I think we might want to add an Identifier#unquote method and use it here as well